### PR TITLE
TS: DeepPartial tooltip options

### PR DESCRIPTION
**Pull request by TorsteinHonsi**
Explicit tooltip options to avoid picks and any-casts.

### CODE REVIEW

File: ts/Core/DefaultOptions.ts [hunk 1]
```diff
@@ -2704,7 +2700,7 @@ const defaultOptions: Options = {
          * @type {Highcharts.ColorString|Highcharts.GradientColorObject|Highcharts.PatternObject}
          */
         backgroundColor: color(palette.neutralColor3)
-            .setOpacity(0.85).get(),
+            .setOpacity(0.85).get() as ColorType,
```
Comment (bre1470) [inline ts/Core/DefaultOptions.ts:2703]: We avoid forced types, because it disables some type checking. Why is this necessary, if `Color.get` returns `Color` (part of ColorType)?

File: ts/Core/Tooltip.ts [hunk 2]
```diff
@@ -145,7 +145,7 @@ class Tooltip {
 
     public now: Record<string, number> = {};
 
-    public options: TooltipOptions = {};
+    public options: TooltipOptions = {} as any;
```
Comment (bre1470) [inline ts/Core/Tooltip.ts:148]: Can we set it here with the defaults? Should the defaults be moved to the class maybe?

**Review by bre1470 — COMMENTED** [review]
Some comments. Otherwise it looks good. 👍

File: ts/Core/DefaultOptions.ts (hunk 1 above)
Comment (TorsteinHonsi) [reply ts/Core/DefaultOptions.ts:2703]: `.get()` apparently returns `ColorType|undefined`...

File: ts/Core/Tooltip.ts (hunk 2 above)
Comment (TorsteinHonsi) [reply ts/Core/Tooltip.ts:148]: Yes probably, but I didn't want to make too many structural changes in this PR.

File: ts/Core/DefaultOptions.ts (hunk 1 above)
Comment (bre1470) [reply ts/Core/DefaultOptions.ts:2703]: Maybe we should stop with `undefined` as an input. It does not sound right to me. An alternative for special cases could be something like a `Color.Null` or `Color.NaC` constant instance of `Color`.

**Review by bre1470 — APPROVED** [review]

File: ts/Core/DefaultOptions.ts (hunk 1 above)
Comment (TorsteinHonsi) [reply ts/Core/DefaultOptions.ts:2703]: I think disallowing `undefined` as a color input makes most sense. I'll make a `@todo` comment about it, so we don't double the scope of this PR.
